pkgs/development/rocm-modules/6/aotriton/default.nix

```diff
@@ -2,226 +2,161 @@
-
-    installPhase = ''
-      runHook preInstall
-      ninja -v install
-      runHook postInstall
-    '';
-
-    cmakeFlags = [
-      "-Wno-dev"
-      "-DAOTRITON_NOIMAGE_MODE=ON" # FIXME: Should be able to build with object code but generate_shim is failing
-      "-DCMAKE_BUILD_TYPE=Release"
-      "-DCMAKE_VERBOSE_MAKEFILE=ON"
-      "-DVIRTUALENV_PYTHON_EXENAME=${lib.getExe py}"
-      "-DCMAKE_CXX_COMPILER=${compiler}"
-      # Manually define CMAKE_INSTALL_<DIR>
-      # See: https://github.com/NixOS/nixpkgs/pull/197838
-      "-DCMAKE_INSTALL_BINDIR=bin"
-      "-DCMAKE_INSTALL_LIBDIR=lib"
-      "-DCMAKE_INSTALL_INCLUDEDIR=include"
-      "-DAMDGPU_TARGETS=${gpuTargets'}"
-      "-DGPU_TARGETS=${gpuTargets'}"
-    ]
-    ++ lib.optionals buildTests [
-      "-DBUILD_CLIENTS_TESTS=ON"
-    ]
-    ++ lib.optionals buildBenchmarks [
-      "-DBUILD_CLIENTS_BENCHMARKS=ON"
-    ]
-    ++ lib.optionals buildSamples [
-      "-DBUILD_CLIENTS_SAMPLES=ON"
-    ];
-
-    postInstall =
-      lib.optionalString buildTests ''
-        mkdir -p $test/bin
-        mv $out/bin/hipblas-test $test/bin
-      ''
-      + lib.optionalString buildBenchmarks ''
-        mkdir -p $benchmark/bin
-        mv $out/bin/hipblas-bench $benchmark/bin
-      ''
-      + lib.optionalString buildSamples ''
-        mkdir -p $sample/bin
-        mv $out/bin/example-* $sample/bin
-      ''
-      + lib.optionalString (buildTests || buildBenchmarks || buildSamples) ''
-        rmdir $out/bin
+  };
+
+  cmakeBuildType = "RelWithDebInfo";
+  separateDebugInfo = true;
+  __structuredAttrs = true;
+  strictDeps = true;
+  # Only set big-parallel when we are building kernels, no-image mode build is faster
+  requiredSystemFeatures = if anySupportedTargets then [ "big-parallel" ] else [ ];
+
+  env.ROCM_PATH = "${clr}";
+  env.CFLAGS = "-w -g1 -gz -Wno-c++11-narrowing";
+  env.CXXFLAGS = finalAttrs.env.CFLAGS;
+
+  nativeBuildInputs = [
+    cmake
+    rocm-cmake
+    pkg-config
+    py
+    ninja
+    clr
+    writableTmpDirAsHomeHook # venv wants to cache in ~
+  ];
+
+  buildInputs = [
+    clr
+    xz
+  ];
+
+  patches = [
+    # CMakeLists.txt: AOTRITON_INHERIT_SYSTEM_SITE_TRITON flag
+    (fetchpatch {
+      sha256 = "sha256-tBmjjhRJmLv3K6F2+4OcMuwf8dH7efPPECMQjh6QdUA=";
+      url = "https://github.com/ROCm/aotriton/commit/9734c3e999c412a07d2b35671998650942b26ed4.patch";
+    })
+  ];
+
+  # Excerpt from README:
+  # Note: do not run ninja separately, due to the limit of the current build system,
+  # ninja install will run the whole build process unconditionally.
+  dontBuild = true;
+  # This builds+installs
+  installPhase = ''
+    runHook preInstall
+    ninja -v install
+    runHook postInstall
+  '';
+  # tests are intended to be ran manually as test/ python scripts and need accelerator
+  doCheck = false;
+  doInstallCheck = false;
+
+  cmakeFlags = [
+    "-Wno-dev"
+    "-DVENV_DIR=/build/venv"
+    "-DVENV_BIN_PYTHON=/build/venv/bin/python"
```

Comment:
Is it possible to paths that don't assume `/build` exists? This will break in `nix-shell`

Reply:
Sadly dropping them breaks the build. Do you have any ideas for an alternate way to set these that will work for nix-shell?